test/e2e/instrumentation/metrics.go

```diff
@@ -44,10 +44,7 @@ var _ = common.SIGDescribe("Metrics", func() {
 		ec = f.KubemarkExternalClusterClientSet
 		gomega.Eventually(ctx, func() error {
 			grabber, err = e2emetrics.NewMetricsGrabber(ctx, c, ec, f.ClientConfig(), true, true, true, true, true, true)
-			if err != nil {
-				framework.ExpectNoError(err, "failed to create metrics grabber")
-			}
-			return nil
+			return err
```

Comment:
This is the right approach. And assertion inside `gomega.Eventually` marks the test as failed, no matter whether it retries.

Reply:
This is a common mistake. I asked about it in the [#ginkgo Slack](https://gophers.slack.com/archives/CQQ50BBNW/p1724666668118819) and suggested a potential solution in https://github.com/nunnatsa/ginkgolinter/issues/127#issuecomment-2495491845.

Reply:
Indeed the fact that it is common is how I was able to recognize it (I have made this mistake many times! :))